internal/terraform/node_resource_abstract_instance.go

```diff
@@ -1173,11 +1126,11 @@ func (n *NodeAbstractResourceInstance) plan(
 		return h.PostDiff(n.Addr, states.CurrentGen, action, priorVal, plannedNewVal)
 	}))
 	if diags.HasErrors() {
-		return plan, state, keyData, diags
+		return nil, nil, keyData, diags
```

Comment:
The predeclared variables were just hiding that we always return `nil`. Since I had to read through all these returns to verify that, I removed them entirely to prevent any more confusion.